### PR TITLE
MOB-1875: Old preview popup for images shared from mobile

### PR DESCRIPTION
**Pull request by AzmiTouil**
Change the activity type to files:spaces to use the new preview.

### CODE REVIEW

**Review by vsellier — COMMENTED** [review]
Does this change will keep the new version of the application compatible with previous versions of PLF ?

**Comment by AzmiTouil** [comment]
it requires a change on integration in the previous versions of PLF, a backport on 5.2.x solves the problem.

**Comment by AzmiTouil** [comment]
To ensure the compatibility of the mobile app with older versions of the platform , we will have to keep the old preview images for <= 5.2.2 versions and use the new one only for the versions strictly > 5.2.2

File: app/src/main/java/org/exoplatform/model/SocialActivity.java [hunk 1]
```diff
@@ -42,7 +45,7 @@
 
   public final static String TYPE_DEFAULT_SPACE = "exosocial:spaces";
 
-  public final static String TYPE_DOC           = "DOC_ACTIVITY";
+  public static String       TYPE_DOC           = "files:spaces";
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/model/SocialActivity.java]: Don't change this constant to a static variable.

You can add constants:
```java
  public final static String       OLD_DOC_ACTIVITY_TYPE           = "files:spaces";

  public final static String       DOC_ACTIVITY_TYPE                     = "DOC_ACTIVITY";

```

File: app/src/main/java/org/exoplatform/service/share/ShareService.java [hunk 2]
```diff
@@ -139,6 +139,9 @@ protected void onHandleIntent(Intent intent) {
    * upload the file
    */
   private UploadInfo initUpload() {
+    int newVersion = PostAction.loadPlatformInformation().platformVersion.substring(0, 5).compareTo("5.2.2");
+    if (newVersion <= 0)
+       SocialActivity.TYPE_DOC = "DOC_ACTIVITY";
     postInfo.type = SocialActivity.TYPE_DOC;
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/service/share/ShareService.java]: Assign here constant `OLD_DOC_ACTIVITY_TYPE` or `DOC_ACTIVITY_TYPE` switch `newVersion` variable.

File: app/src/main/java/org/exoplatform/service/share/PostAction.java [hunk 3]
```diff
@@ -86,6 +87,22 @@ private void setPostTextActivity() {
       postInfo.type = SocialActivity.TYPE_DEFAULT_SPACE;
   }
 
+  public static SocialRestService.PlatformInformation loadPlatformInformation() {
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/service/share/PostAction.java]: The platform server version is already retrieved here:
https://github.com/exoplatform/exo-android/blob/fix/MOB-1875/app/src/main/java/org/exoplatform/tool/ServerUtils.java#L101

**Review by boubaker — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/exoplatform/model/SocialActivity.java (hunk 1 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/model/SocialActivity.java]: done

File: app/src/main/java/org/exoplatform/service/share/ShareService.java (hunk 2 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/service/share/ShareService.java]: done

File: app/src/main/java/org/exoplatform/service/share/PostAction.java (hunk 3 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/service/share/PostAction.java]: Okay,I'll use it

File: app/src/main/java/org/exoplatform/activity/ShareExtensionActivity.java [hunk 4]
```diff
@@ -123,7 +123,9 @@ protected void onCreate(Bundle bundle) {
       } else {
         // User is sharing some document(s)
         mActivityPost.title = "";
-        mActivityPost.type = SocialActivity.TYPE_DOC;
+        if (ServerUtils.isOldVersion())
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/activity/ShareExtensionActivity.java]: Use braces using if / else block, (see DEVSTD-17)

File: app/src/main/java/org/exoplatform/fragment/ComposeFragment.java [hunk 5]
```diff
@@ -161,7 +162,10 @@ public void onResume() {
     setSpaceLabel();
     setTouchListener();
     SocialActivity activity = getShareActivity().getActivityPost();
-    if (!SocialActivity.TYPE_DOC.equals(activity.type)) {
+    String type = SocialActivity.DOC_ACTIVITY_TYPE;
+    if (ServerUtils.isOldVersion())
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/fragment/ComposeFragment.java]: Here too (See DEVSTD-17)

File: app/src/main/java/org/exoplatform/service/share/PostAction.java [hunk 6]
```diff
@@ -61,8 +62,10 @@ public static SocialActivity execute(SocialActivity post, PostActionListener lis
 
   @Override
   protected boolean doExecute() {
-
-    if (SocialActivity.TYPE_DOC.equals(postInfo.type)) {
+    String type = SocialActivity.DOC_ACTIVITY_TYPE;
+    if (ServerUtils.isOldVersion())
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/service/share/PostAction.java]: Here too (See DEVSTD-17)

File: app/src/main/java/org/exoplatform/service/share/ShareService.java [hunk 7]
```diff
@@ -139,7 +140,9 @@ protected void onHandleIntent(Intent intent) {
    * upload the file
    */
   private UploadInfo initUpload() {
-    postInfo.type = SocialActivity.TYPE_DOC;
+    if (ServerUtils.isOldVersion())
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/service/share/ShareService.java]: Here too (See DEVSTD-17)

File: app/src/main/java/org/exoplatform/tool/ServerUtils.java [hunk 8]
```diff
@@ -62,6 +62,14 @@
 
   public static final String LOG_TAG = "ServerUtils";
 
+  public static String plfVersion;
+
+  public static boolean isOldVersion(){
+    if (plfVersion.substring(0, 5).compareTo("5.2.2")<=0)
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/tool/ServerUtils.java]: code formatted (DEVSTD-4 and DEVSTD-5)

File: app/src/main/java/org/exoplatform/tool/ServerUtils.java [hunk 9]
```diff
@@ -98,8 +106,8 @@ public void onResponse(Call<PlatformInfo> call, Response<PlatformInfo> response)
                 // Update the server URL to the final value
                 server.setUrl(finalUrl);
               } catch (MalformedURLException ignored) {}
-              Double plfVersion = ServerUtils.convertVersionFromString(response.body().platformVersion);
-              if (plfVersion >= App.Platform.MIN_SUPPORTED_VERSION) {
+              plfVersion = response.body().platformVersion.substring(0, 5);
```
Comment (boubaker) [inline app/src/main/java/org/exoplatform/tool/ServerUtils.java]: why this substring ?

File: app/src/main/java/org/exoplatform/activity/ShareExtensionActivity.java (hunk 4 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/activity/ShareExtensionActivity.java]: Okay

File: app/src/main/java/org/exoplatform/tool/ServerUtils.java (hunk 9 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/tool/ServerUtils.java]: it's fault ,I'll delete it

File: app/src/main/java/org/exoplatform/activity/ShareExtensionActivity.java (hunk 4 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/activity/ShareExtensionActivity.java]: done

File: app/src/main/java/org/exoplatform/fragment/ComposeFragment.java (hunk 5 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/fragment/ComposeFragment.java]: Done

File: app/src/main/java/org/exoplatform/service/share/PostAction.java (hunk 6 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/service/share/PostAction.java]: Done

File: app/src/main/java/org/exoplatform/service/share/ShareService.java (hunk 7 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/service/share/ShareService.java]: Done

File: app/src/main/java/org/exoplatform/tool/ServerUtils.java (hunk 8 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/tool/ServerUtils.java]: Done

File: app/src/main/java/org/exoplatform/tool/ServerUtils.java (hunk 9 above)
Comment (AzmiTouil) [reply app/src/main/java/org/exoplatform/tool/ServerUtils.java]: removed

**Review by boubaker — APPROVED** [review]

**Pull request merged**